File: src/ChpokkWeb/SystemFiles/Templates/ProjectTemplates/VisualBasic/Silverlight/1049/BusinessApplication/BA.Web.10.0/Models/Shared/User.shared.vb
Namespace Web

    Partial Public Class User
        ''' <summary>
        ''' Возвращает отображаемое имя пользователя, которое по умолчанию равно значению свойства FriendlyName.
        ''' Если свойство FriendlyName не задано, возвращается имя пользователя.
        ''' </summary>
        Public ReadOnly Property DisplayName() As String
            Get
                If Not String.IsNullOrEmpty(Me.FriendlyName) Then
                    Return Me.FriendlyName
                Else
                    Return Me.Name
                End If
            End Get
        End Property
    End Class
End Namespace

File: src/ChpokkWeb/SystemFiles/Templates/ProjectTemplates/CSharp/Silverlight/1036/BusinessApplication/BA.Web.10.0/Models/Shared/User.shared.cs
namespace $safeprojectname$
{
    /// <summary>
    /// Classe partielle étendant le type User qui ajoute des propriétés et des méthodes partagées.
    /// Ces propriétés et méthodes seront disponibles sur l'application serveur et sur l'application cliente.
    /// </summary>
    public partial class User
    {
        /// <summary>
        /// Retourne le nom complet de l'utilisateur, qui est par défaut son FriendlyName.
        /// Si FriendlyName n'est pas défini, le nom d'utilisateur est retourné.
        /// </summary>
        public string DisplayName
        {
            get
            {
                if (!string.IsNullOrEmpty(this.FriendlyName))
                {
                    return this.FriendlyName;
                }
                else
                {
                    return this.Name;
                }
            }
        }
    }
}
